blog/2024-10-29-vibration-analysis-python/index.mdx

```diff
@@ -0,0 +1,376 @@
+
+plt.subplot(1, 4, 4)
+plt.plot(high_kurtosis_signal, c="indigo")
+plt.title(f"High Kurtosis: {high_kurtosis:.2f}")
+
+plt.show()
+```
+
+![Kurtosis plot](./img/kurtosis.png)
+
+Here we see kurtosis values for square, sine, zero-centered normally distributed noise with unit standard deviation (which results in a value near 3), and the same noise with added high peaks, that affect a tail of the distribution and increase kurtosis.
+
+**Skewness**, the 4th central moment, reflects asymmetry in amplitudes, which in turn may be caused by any mechanical asymmetry like rubbing of rotating parts, specific wear, etc. As said above, it may be hard to interpret skewness in acceleration in terms of mechanics, and for some mechanical systems skewness may be nonzero by design, so in most cases it should be analyzed in comparison to the proven reference normal state of the machine. Also, skewness as a measure of directed acceleration impulses, may strongly depend on the orientation of the sensor. With this directional selectivity, it may be useful for discovering subtle, but meaningful changes in operation.
+
+in Python, skewness can be found as follows:
+
+```python
+import numpy as np
+import matplotlib.pyplot as plt
+
+np.random.seed(0)
+positive_skew_signal = np.random.exponential(1, 1000) - 0.5
+near_centered_signal = np.random.randn(1000) + np.sin(2 * np.pi * 5 * np.linspace(0, 1, 1000))
+negative_skew_signal = -positive_skew_signal
+
+def skewness(window):
+    return np.mean((window - np.mean(window)) ** 3) / (np.std(window) ** 3)
+
+positive_skewness = skewness(positive_skew_signal)
+near_zero_skewbess = skewness(near_centered_signal)
+negative_skewness = skewness(negative_skew_signal)
+
+plt.figure(figsize=(12, 4))
+
+plt.subplot(1, 3, 1)
+plt.plot(positive_skew_signal, c="indigo")
+plt.title(f"Positive Skewness: {positive_skewness:.2f}")
+
+plt.subplot(1, 3, 2)
+plt.plot(near_centered_signal, c="indigo")
+plt.title(f"Near-zero Skewness: {near_zero_skewbess:.2f}")
+
+plt.subplot(1, 3, 3)
+plt.plot(negative_skew_signal, c="indigo")
+plt.title(f"Negative Skewness: {negative_skewness:.2f}")
+
+plt.show()
+```
+
+![Skewness plot](./img/skweness.png)
+
+As expected, for the signal with the zero-symmetric distribution of the amplitude (5 full periods of noisy sine wave), skewness is about 0.
+
+### Amplitude spectrum, spectrogram, power spectral density
+
+One of the ways to get a detailed view of the vibration signal is spectral analysis. Roughly speaking, the spectrum is a decomposition of the signal by elementary sine components of different frequencies, so that the signal, originally represented as a function of time (in the time domain), gets represented as a function of frequency - in the frequency domain.
+
+The procedure of conversion, well known as the Fourier transform, possesses numerous fascinatingly powerful mathematical properties, including
+
+- linearity; the transform decomposes the signal into linear combinations of its components,
```

Comment:
Formatting to check:

- **linearity**: 
- **duality**:
- ..

Reply:
replaces ; with :